src/components/NavBar/NavMenu.jsx

```diff
@@ -52,7 +53,12 @@ const NavMenu = ({
   const theme = useTheme();
   const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'));
   const { setContact } = useContext(DocumentListContext);
-  const { numUnreadMessages } = useContext(MessageContext);
+  const { data } = useMessageList('Inbox');
+  const [numUnreadMessages, setNumUnreadMessages] = useState(0);
+
+  useEffect(() => {
```

Comment:
You can compute the number of unread messages in the the `useMessageList` hook, or derive it directly where you display it in the component based on current state. Then you don't need to add this `useEffect` in consuming components, and you don't need to worry about the value in `numUnreadMessages` falling out of sync with what's in the hook.

Reply:
Ah, okay. Just tried it out without the `useEffect` and directly computing `numUnreadMessages` with:

```
const numUnreadMessages = data?.reduce((a, m) => (!m.readStatus ? a + 1 : a), 0);
```

https://github.com/codeforpdx/PASS/assets/14917816/fe93762a-f58b-4d65-b48c-9ce1760bb4b0

I'll replace the one in `NavDesktop.jsx` as well.